### ISSUE TITLE
not sure if bug

### ISSUE THREAD

**Issue by StempelLab** [talmolab/sleap#2560]
not sure if this is a bug or installation issue. but when i try to train an existing .slp project with SLEAP 1,5 or 1.6 - source version, i see no errors but the training never starts.
this is what my terminal looks like:

Software versions:
SLEAP: 1.5.2
Numpy: 2.3.5
Python: 3.12.11
OS: Windows-10-10.0.19045-SP0

Happy SLEAPing! :)
Resetting monitor window.
Polling: //gpfs.corp.brain.mpg.de/stem/data/project_hierarchy/Sleap_projects/mini2p_bottom/models/260113_105931.single_instance.n=1274\viz\validation.*.png
Start training single_instance...
['sleap-nn-train', '--config-name', '260113_105932_config', '--config-dir', 'C:\\Users\\SU-WEI~1\\AppData\\Local\\Temp\\tmp_43nzsbc']
Input config:

data_config:
  train_labels_path:
  - //gpfs.corp.brain.mpg.de/stem/data/project_hierarchy/Sleap_projects/mini2p_bottom/bottom_2026.slp
  val_labels_path: null
  validation_fraction: 0.3
  test_file_path: null
  provider: LabelsReader
  user_instances_only: true
  data_pipeline_fw: torch_dataset_cache_img_memory
  cache_img_path: null
  use_existing_imgs: false
  delete_cache_imgs_after_training: true
  preprocessing:
    ensure_rgb: false
    ensure_grayscale: false
    max_height: 1920
    max_width: 1920
    scale: 0.45
    crop_size: null
    min_crop_size: 100
  use_augmentations_train: true
  augmentation_config:
    intensity:
      uniform_noise_min: 0.0
      uniform_noise_max: 1.0
      uniform_noise_p: 0.0
      gaussian_noise_mean: 5.0
      gaussian_noise_std: 1.0
      gaussian_noise_p: 0.0
      contrast_min: 0.5
      contrast_max: 2.0
      contrast_p: 0.0
      brightness_min: 0.0
      brightness_max: 2.0
      brightness_p: 0.0
    geometric:
      rotation_min: -180.0
      rotation_max: 180.0
      scale_min: 0.9
      scale_max: 1.1
      translate_width: 0.0
      translate_height: 0.0
      affine_p: 1.0
      erase_scale_min: 0.0001
      erase_scale_max: 0.01
      erase_ratio_min: 1.0
      erase_ratio_max: 1.0
      erase_p: 0.0
      mixup_lambda_min: 0.01
      mixup_lambda_max: 0.05
      mixup_p: 0.0
  skeletons: []
model_config:
  init_weights: default
  pretrained_backbone_weights: null
  pretrained_head_weights: null
  backbone_config:
    unet:
      in_channels: 1
      kernel_size: 3
      filters: 16
      filters_rate: 2.0
      max_stride: 16
      stem_stride: null
      middle_block: true
      up_interpolate: true
      stacks: 1
      convs_per_block: 2
      output_stride: 2
    convnext: null
    swint: null
  head_configs:
    single_instance:
      confmaps:
        part_names:
        - nose
        - left_paw
        - right_paw
        - left_foot
        - 'right_foot '
        - tail_base
        - belly
        sigma: 2.5
        output_stride: 2
    centroid: null
    centered_instance: null
    bottomup: null
    multi_class_bottomup: null
    multi_class_topdown: null
  total_params: null
trainer_config:
  train_data_loader:
    batch_size: 50
    num_workers: 0
    shuffle: true
  val_data_loader:
    batch_size: 50
    num_workers: 0
    shuffle: false
  model_ckpt:
    save_top_k: 1
    save_last: false
  trainer_devices: null
  trainer_device_indices: null
  trainer_accelerator: cuda
  profiler: null
  trainer_strategy: auto
  enable_progress_bar: true
  min_train_steps_per_epoch: 200
  train_steps_per_epoch: 200
  visualize_preds_during_training: true
  keep_viz: false
  max_epochs: 100
  seed: null
  use_wandb: false
  save_ckpt: true
  ckpt_dir: //gpfs.corp.brain.mpg.de/stem/data/project_hierarchy/Sleap_projects/mini2p_bottom/models
  run_name: 260113_105931.single_instance.n=1274
  resume_ckpt_path: null
  wandb:
    entity: ''
    project: ''
    name: ''
    save_viz_imgs_wandb: false
    api_key: ''
    wandb_mode: null
    prv_runid: ''
    group: ''
    current_run_id: null
  optimizer_name: Adam
  optimizer:
    lr: 0.0005
    amsgrad: false
  lr_scheduler:
    step_lr: null
    reduce_lr_on_plateau:
      threshold: 1.0e-06
      threshold_mode: abs
      cooldown: 3
      patience: 5
      factor: 0.5
      min_lr: 1.0e-08
  early_stopping:
    min_delta: 1.0e-08
    patience: 10
    stop_training_on_plateau: true
  online_hard_keypoint_mining:
    online_mining: false
    hard_to_easy_ratio: 2.0
    min_hard_keypoints: 1
    max_hard_keypoints: 5
    loss_scale: 5.0
  zmq:
    controller_port: 9000
    controller_polling_timeout: 10
    publish_port: 9001
name: ''
description: ''
sleap_nn_version: 0.0.5
filename: ''

2026-01-13 11:00:46 | INFO | sleap_nn.train:run_training:24 | Started training at: 2026-01-13 11:00:46.279667
2026-01-13 11:00:53 | INFO | sleap_nn.training.model_trainer:_setup_train_val_labels:216 | Creating train-val split...
2026-01-13 11:02:14 | INFO | sleap_nn.training.model_trainer:_setup_train_val_labels:261 | # Train Labeled frames: 891
2026-01-13 11:02:14 | INFO | sleap_nn.training.model_trainer:_setup_train_val_labels:262 | # Val Labeled frames: 382
2026-01-13 11:02:14 | INFO | sleap_nn.training.model_trainer:setup_config:510 | Setting up config...
2026-01-13 11:02:16 | INFO | sleap_nn.training.model_trainer:train:849 | Setting up for training...
2026-01-13 11:02:16 | INFO | sleap_nn.training.model_trainer:_setup_model_ckpt_dir:573 | Setting up model ckpt dir: `//gpfs.corp.brain.mpg.de/stem/data/project_hierarchy/Sleap_projects/mini2p_bottom/models/260113_105931.single_instance.n=1274`...
2026-01-13 11:02:38 | INFO | sleap_nn.training.model_trainer:train:864 | Setting up visualization train and val datasets...
2026-01-13 11:02:38 | INFO | sleap_nn.training.model_trainer:train:868 | Setting up Trainer...
2026-01-13 11:02:38 | INFO | sleap_nn.training.model_trainer:_setup_loggers_callbacks:645 | Setting up callbacks and loggers...
2026-01-13 11:02:38 | INFO | sleap_nn.training.callbacks:__init__:228 | Training controller subscribed to: tcp://127.0.0.1:9000 (topic: )
2026-01-13 11:02:38 | INFO | sleap_nn.training.callbacks:__init__:287 | ProgressReporterZMQ publishing to tcp://127.0.0.1:9001 for ''
2026-01-13 11:02:38 | INFO | sleap_nn.training.model_trainer:train:897 | Trainer devices: auto
INFO: GPU available: True (cuda), used: True
INFO:lightning.pytorch.utilities.rank_zero:GPU available: True (cuda), used: True
INFO: TPU available: False, using: 0 TPU cores
INFO:lightning.pytorch.utilities.rank_zero:TPU available: False, using: 0 TPU cores

**Comment by gitttt-1234** [comment]
Hi @StempelLab !

This could be the memory caching step. We added a progress bar for this in the latest `sleap-nn` prerelease - could you try upgrading to confirm `uv pip install --prerelease allow sleap --extra nn`? The caching happens before training begins and can take a moment depending on dataset size.

Let us know if you have any questions!

Thanks,

Divya

**Comment by gitttt-1234** [comment]
closing this issue, feel free to re-open if you have any questions!
